pkg/customresource/repositoryserver.yaml

```diff
@@ -0,0 +1,278 @@
+                        type: string
+                    type: object
+                  rootPath:
+                    type: string
+                  username:
+                    type: string
+                required:
+                - hostname
+                - passwordSecretRef
+                - rootPath
+                - username
+                type: object
+              server:
+                description: Server details required for starting the repository proxy
+                  server and initializing the repository client users
+                properties:
+                  adminSecretRef:
+                    description: SecretReference represents a Secret Reference. It
+                      has enough information to retrieve secret in any namespace
+                    properties:
+                      name:
+                        description: name is unique within a namespace to reference
+                          a secret resource.
+                        type: string
+                      namespace:
+                        description: namespace defines the space within which the
+                          secret name must be unique.
+                        type: string
+                    type: object
+                  tlsSecretRef:
+                    description: SecretReference represents a Secret Reference. It
+                      has enough information to retrieve secret in any namespace
+                    properties:
+                      name:
+                        description: name is unique within a namespace to reference
+                          a secret resource.
+                        type: string
+                      namespace:
+                        description: namespace defines the space within which the
+                          secret name must be unique.
+                        type: string
+                    type: object
+                  userAccess:
+                    properties:
+                      userAccessSecretRef:
+                        description: SecretReference represents a Secret Reference.
+                          It has enough information to retrieve secret in any namespace
+                        properties:
+                          name:
+                            description: name is unique within a namespace to reference
+                              a secret resource.
+                            type: string
+                          namespace:
+                            description: namespace defines the space within which
+                              the secret name must be unique.
+                            type: string
+                        type: object
+                        x-kubernetes-map-type: atomic
+                      username:
+                        type: string
+                    required:
+                    - userAccessSecretRef
+                    - username
+                    type: object
```

Comment:
Server users should be an array instead of a single object right?

Reply:
Oh, as per https://github.com/kanisterio/kanister/pull/1653/files#diff-8e1bee7434fbff0335bff585b3dda95d4c382937dc0e7086bb51bdf5c2738085R138-R141, understood userAccessSecretRef would contain
`"hostname":"password"`
key value pair. Right?

Reply:
yes correct